### ISSUE TITLE
Cargo test fails on Apple Silicon "the `att_syntax` option is only supported on x86 - ui/asm_syntax.rs"

### ISSUE THREAD

**Issue by basavesh** [rust-lang/rust-clippy#7091]
I tried this code:

```rust
cargo test
```

I expected to see this happen: *test [ui] ui/asm_syntax.rs ... ok*

Instead, this happened: *test compile_test ... FAILED*

### Meta

- `git commit`:52c8c9c520359f197b15aa6b69e7a779c9be8eb5
- `rustc -Vv`:
  ```
  rustc 1.53.0-nightly (361bfce30 2021-04-07)
  binary: rustc
  commit-hash: 361bfce305b8829eda7f3d133fe82a118685de1f
  commit-date: 2021-04-07
  host: aarch64-apple-darwin
  release: 1.53.0-nightly
  LLVM version: 12.0.0
  ```

<details><summary>Backtrace</summary>
  <p>
  
  ```
failures:

---- compile_test stdout ----
diff of stderr:

+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:10:9
+   |
+LL |         asm!("", options(att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:11:9
+   |
+LL |         asm!("", options(nostack, att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:21:9
+   |
+LL |         asm!("", options(att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: the `att_syntax` option is only supported on x86
+  --> $DIR/asm_syntax.rs:22:9
+   |
+LL |         asm!("", options(nostack, att_syntax));
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
 error: Intel x86 assembly syntax used
   --> $DIR/asm_syntax.rs:7:9
    |
 LL |         asm!("");
    |         ^^^^^^^^^
    |
    = note: `-D clippy::inline-asm-x86-intel-syntax` implied by `-D warnings`
    = help: use AT&T x86 assembly syntax
 
 error: Intel x86 assembly syntax used
   --> $DIR/asm_syntax.rs:8:9
    |
 LL |         asm!("", options());
    |         ^^^^^^^^^^^^^^^^^^^^
    |
    = help: use AT&T x86 assembly syntax
 
 error: Intel x86 assembly syntax used
   --> $DIR/asm_syntax.rs:9:9
    |
 LL |         asm!("", options(nostack));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: use AT&T x86 assembly syntax
 
 error: AT&T x86 assembly syntax used
   --> $DIR/asm_syntax.rs:21:9
    |
 LL |         asm!("", options(att_syntax));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D clippy::inline-asm-x86-att-syntax` implied by `-D warnings`
    = help: use Intel x86 assembly syntax
 
 error: AT&T x86 assembly syntax used
   --> $DIR/asm_syntax.rs:22:9
    |
 LL |         asm!("", options(nostack, att_syntax));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: use Intel x86 assembly syntax
 
-error: aborting due to 5 previous errors
+error: aborting due to 9 previous errors
 
 

The actual stderr differed from the expected stderr.
Actual stderr saved to /Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/test_build_base/asm_syntax.stage-id.stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args asm_syntax.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: "/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/clippy-driver" "tests/ui/asm_syntax.rs" "-L" "/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/test_build_base" "--target=aarch64-apple-darwin" "--error-format" "json" "-C" "prefer-dynamic" "-o" "/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/test_build_base/asm_syntax.stage-id" "-A" "unused" "--emit=metadata" "-L" "/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps" "-L" "/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps" "-Dwarnings" "-Zui-testing" "--extern" "syn=/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps/libsyn-5eb42541658f69e7.rlib" "--extern" "regex=/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps/libregex-c84e0e69e77f9f25.rlib" "--extern" "quote=/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps/libquote-8979ab63a16019f5.rlib" "--extern" "serde=/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps/libserde-87d3a021f90b619c.rlib" "--extern" "clippy_lints=/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/deps/libclippy_lints-4f4ac6c6ca75663e.rlib" "-L" "/Users/basaveshshivakumar/research/rust-clippy-upstream/target/debug/test_build_base/asm_syntax.stage-id.aux"
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
{"message":"the `att_syntax` option is only supported on x86","code":null,"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":223,"byte_end":253,"line_start":10,"line_end":10,"column_start":9,"column_end":39,"is_primary":true,"text":[{"text":"        asm!(\"\", options(att_syntax));","highlight_start":9,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: the `att_syntax` option is only supported on x86\n  --> tests/ui/asm_syntax.rs:10:9\n   |\nLL |         asm!(\"\", options(att_syntax));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
{"message":"the `att_syntax` option is only supported on x86","code":null,"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":262,"byte_end":301,"line_start":11,"line_end":11,"column_start":9,"column_end":48,"is_primary":true,"text":[{"text":"        asm!(\"\", options(nostack, att_syntax));","highlight_start":9,"highlight_end":48}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: the `att_syntax` option is only supported on x86\n  --> tests/ui/asm_syntax.rs:11:9\n   |\nLL |         asm!(\"\", options(nostack, att_syntax));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
{"message":"the `att_syntax` option is only supported on x86","code":null,"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":497,"byte_end":527,"line_start":21,"line_end":21,"column_start":9,"column_end":39,"is_primary":true,"text":[{"text":"        asm!(\"\", options(att_syntax));","highlight_start":9,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: the `att_syntax` option is only supported on x86\n  --> tests/ui/asm_syntax.rs:21:9\n   |\nLL |         asm!(\"\", options(att_syntax));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
{"message":"the `att_syntax` option is only supported on x86","code":null,"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":536,"byte_end":575,"line_start":22,"line_end":22,"column_start":9,"column_end":48,"is_primary":true,"text":[{"text":"        asm!(\"\", options(nostack, att_syntax));","highlight_start":9,"highlight_end":48}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":"error: the `att_syntax` option is only supported on x86\n  --> tests/ui/asm_syntax.rs:22:9\n   |\nLL |         asm!(\"\", options(nostack, att_syntax));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
{"message":"Intel x86 assembly syntax used","code":{"code":"clippy::inline_asm_x86_intel_syntax","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":140,"byte_end":149,"line_start":7,"line_end":7,"column_start":9,"column_end":18,"is_primary":true,"text":[{"text":"        asm!(\"\");","highlight_start":9,"highlight_end":18}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`-D clippy::inline-asm-x86-intel-syntax` implied by `-D warnings`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"use AT&T x86 assembly syntax","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: Intel x86 assembly syntax used\n  --> tests/ui/asm_syntax.rs:7:9\n   |\nLL |         asm!(\"\");\n   |         ^^^^^^^^^\n   |\n   = note: `-D clippy::inline-asm-x86-intel-syntax` implied by `-D warnings`\n   = help: use AT&T x86 assembly syntax\n\n"}
{"message":"Intel x86 assembly syntax used","code":{"code":"clippy::inline_asm_x86_intel_syntax","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":158,"byte_end":178,"line_start":8,"line_end":8,"column_start":9,"column_end":29,"is_primary":true,"text":[{"text":"        asm!(\"\", options());","highlight_start":9,"highlight_end":29}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"use AT&T x86 assembly syntax","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: Intel x86 assembly syntax used\n  --> tests/ui/asm_syntax.rs:8:9\n   |\nLL |         asm!(\"\", options());\n   |         ^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: use AT&T x86 assembly syntax\n\n"}
{"message":"Intel x86 assembly syntax used","code":{"code":"clippy::inline_asm_x86_intel_syntax","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":187,"byte_end":214,"line_start":9,"line_end":9,"column_start":9,"column_end":36,"is_primary":true,"text":[{"text":"        asm!(\"\", options(nostack));","highlight_start":9,"highlight_end":36}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"use AT&T x86 assembly syntax","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: Intel x86 assembly syntax used\n  --> tests/ui/asm_syntax.rs:9:9\n   |\nLL |         asm!(\"\", options(nostack));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: use AT&T x86 assembly syntax\n\n"}
{"message":"AT&T x86 assembly syntax used","code":{"code":"clippy::inline_asm_x86_att_syntax","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":497,"byte_end":527,"line_start":21,"line_end":21,"column_start":9,"column_end":39,"is_primary":true,"text":[{"text":"        asm!(\"\", options(att_syntax));","highlight_start":9,"highlight_end":39}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"`-D clippy::inline-asm-x86-att-syntax` implied by `-D warnings`","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"use Intel x86 assembly syntax","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: AT&T x86 assembly syntax used\n  --> tests/ui/asm_syntax.rs:21:9\n   |\nLL |         asm!(\"\", options(att_syntax));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = note: `-D clippy::inline-asm-x86-att-syntax` implied by `-D warnings`\n   = help: use Intel x86 assembly syntax\n\n"}
{"message":"AT&T x86 assembly syntax used","code":{"code":"clippy::inline_asm_x86_att_syntax","explanation":null},"level":"error","spans":[{"file_name":"tests/ui/asm_syntax.rs","byte_start":536,"byte_end":575,"line_start":22,"line_end":22,"column_start":9,"column_end":48,"is_primary":true,"text":[{"text":"        asm!(\"\", options(nostack, att_syntax));","highlight_start":9,"highlight_end":48}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"use Intel x86 assembly syntax","code":null,"level":"help","spans":[],"children":[],"rendered":null}],"rendered":"error: AT&T x86 assembly syntax used\n  --> tests/ui/asm_syntax.rs:22:9\n   |\nLL |         asm!(\"\", options(nostack, att_syntax));\n   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n   |\n   = help: use Intel x86 assembly syntax\n\n"}
{"message":"aborting due to 9 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to 9 previous errors\n\n"}

------------------------------------------

thread 'compile_test' panicked at 'Some tests failed', /Users/basaveshshivakumar/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.6.0/src/lib.rs:105:22
stack backtrace:
   0: std::panicking::begin_panic
             at /Users/basaveshshivakumar/.rustup/toolchains/nightly-2021-04-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/std/src/panicking.rs:519:12
   1: compiletest_rs::run_tests
             at /Users/basaveshshivakumar/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.6.0/src/lib.rs:105:22
   2: compile_test::run_mode
             at ./tests/compile-test.rs:101:5
   3: compile_test::compile_test
             at ./tests/compile-test.rs:270:5
   4: compile_test::compile_test::{{closure}}
             at ./tests/compile-test.rs:267:1
   5: core::ops::function::FnOnce::call_once
             at /Users/basaveshshivakumar/.rustup/toolchains/nightly-2021-04-08-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/ops/function.rs:227:5
   6: core::ops::function::FnOnce::call_once
             at /rustc/361bfce305b8829eda7f3d133fe82a118685de1f/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    compile_test

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 10.34s

error: test failed, to rerun pass '--test compile-test'
  ```
  
  </p>
</details>
